storage/storage_transport.go

```diff
@@ -0,0 +1,77 @@
+package storage
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/storage"
+)
+
+var (
+	// Transport is an ImageTransport that uses a default storage.Mall or
+	// one that's it's explicitly told to use.
+	Transport MallTransport = &storageTransport{}
+	// ErrUnknownPolicyConfigurationScope is returned when
+	// ValidatePolicyConfigurationScope() fails.
+	ErrUnknownPolicyConfigurationScope = errors.New("unknown policy configuration scope")
+	// ErrInvalidReference is returned when ParseReference() is passed an
+	// empty reference.
+	ErrInvalidReference = errors.New("invalid reference")
+	idRegexp            = regexp.MustCompile("^(sha256:)?([0-9a-fA-F]{64})$")
+)
+
+// MallTransport is an ImageTransport that uses a storage.Mall, either its own
+// or one that it's told to use.
+type MallTransport interface {
+	types.ImageTransport
+	SetMall(storage.Mall) error
+}
+
+type storageTransport struct {
+	mall storage.Mall
+}
+
+func (s *storageTransport) Name() string {
+	return "oci-storage"
```

Comment:
If there is a clear concept “the default” Mall instance, and users would only use a different one for debugging or other corner cases, it might make sense to only support references to that the default mall, and allow overriding it only via a `types.SystemContext` as introduced in #52 .